### PR TITLE
Fixes for regression in the persistency of std::string payloads

### PR DESCRIPTION
**Pull request by ggovi**
Fix for std::string type matching.  A recent compiler change has changed the output of the 'demangler' for the std::string class.
Regression test adapted for execution in IB environment. This test will now run correctly once the above fix will be in a main release

### CODE REVIEW

**Comment by ggovi** [comment]
superseding #18542
@smuzaffar 
this one contains fixes for the std::string persistency ( the demangled name from type_info.name() has changed ) and also the adaptation of the regression test for the IB environment. The execution of the test is fine ( no crash ), but the output will be ok only once the std::string fix reaches the main release referenced by the IBs. 
Please let me know if we should disable this test until the fix get in the main release.

**Comment by ggovi** [comment]
please test

File: CondCore/CondDB/test/condTestRegression.py [hunk 1]
```diff
@@ -160,10 +157,24 @@ def runSelf(self, readOrWrite, dbNameIn=None):
...
 
+          execName = 'test/%s/testReadWritePayloads' %self.arch
+          executable = '%s/%s' %(os.environ['LOCALRT'],execName)
+          if not os.path.exists(executable):
+              print 'Executable %s not found in local release.'
+              executable = None
+              for rel_base_env in ['CMSSW_BASE', 'CMSSW_RELEASE_BASE', 'CMSSW_FULL_RELEASE_BASE' ]:
+                  #print "Looking for: %s = %s " %(rel_base_env,os.getenv(rel_base_env))
+                  if os.getenv(rel_base_env) and os.path.exists(str(os.environ[rel_base_env])+'/%s' %execName):
+                      executable = str(os.environ[rel_base_env])+'/%s' %execName
```
Comment (smuzaffar) [inline CondCore/CondDB/test/condTestRegression.py]: @ggovi , you should break the loop once executable is found

Comment (ggovi) [inline CondCore/CondDB/test/condTestRegression.py]: oops, you're right, thanks.

**Comment by smuzaffar** [comment]
please test

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by ggovi** [comment]
@smuzaffar 
this will fail in the IB environment, until the fixes reaches a main release... is it ok?

**Comment by smuzaffar** [comment]
@ggovi , which main release?

**Comment by ggovi** [comment]
@smuzaffar 
well, the release that the IBs are referencing , in this case to find the executable testReadWritePayloads

**Comment by davidlange6** [comment]
Hi @ggovi - i would have thought the logic  in CondCore/CondDB/test/condTestRegression.py would have been sufficient to find testReadWritePayloads locally (after scram b)

**Comment by smuzaffar** [comment]
@ggovi , I do not see it failing in PR tests and my local tests so looks like it was able to find testReadWritePayloads from the release area. So it looks good to me

```
-rwxr-xr-x 1 cvmfs cvmfs 179632 Aug 25  2016 /cvmfs/cms.cern.ch/slc6_amd64_gcc493/cms/cmssw/CMSSW_7_6_6/test/slc6_amd64_gcc493/testReadWritePayloads
-rwxr-xr-x 1 cvmfs cvmfs 177600 Dec  2 02:38 /cvmfs/cms.cern.ch/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_0/test/slc6_amd64_gcc530/testReadWritePayloads
-rwxr-xr-x 1 cvmfs cvmfs 193408 Apr  9 02:27 /cvmfs/cms.cern.ch/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_1/test/slc6_amd64_gcc530/testReadWritePayloads
-rwxr-xr-x 1 cvmfs cvmfs 177608 Feb  2 01:43 /cvmfs/cms.cern.ch/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_0_26/test/slc6_amd64_gcc530/testReadWritePayloads
```

**Comment by ggovi** [comment]
@davidlange6 @smuzaffar 
yes ,that's the idea. Maybe I'm confused then - in what circumstance the IB is not building the test?

**Comment by smuzaffar** [comment]
@ggovi , test fails in IB env where the ```$LOCALRT/test/$SCRAM_ARCH/testReadWritePayloads``` does not exist but you change here https://github.com/cms-sw/cmssw/pull/18579/files#diff-44fb5ff277a0e30437f3b786451c29c7R161 should fix it.

**Comment by davidlange6** [comment]
+1

**Pull request merged**